main/src/main/scala/sbt/internal/server/LanguageServerProtocol.scala

```diff
@@ -80,6 +81,37 @@ private[sbt] object LanguageServerProtocol {
               import sbt.protocol.codec.JsonProtocol._
               val param = Converter.fromJson[Q](json(r)).get
               onSettingQuery(Option(r.id), param)
+            case r: JsonRpcRequestMessage if r.method == "sbt/cancelRequest" =>
+              val param = Converter.fromJson[CancelRequestParams](json(r)).get
+              try {
+                val (state, runningEngine) = EvaluateTask.currentlyRunningEngine.get
```

Comment:
This could return `null`, so maybe put this in `Option(EvaluateTask.currentlyRunningEngine.get)`?

Reply:
actually this is expected, any `NonFatal` error (like NPE) that happens within the try block is catch and return to the user a proper error  answer.

Reply:
there are more unsafe get also in the following lines, do you prefer to handle all of them one-by-one?

Reply:
However you want to handle it is fine, but we should handle them.